extensions/extensions-helm/extensions-helm-spark-infrastructure/aissemble-hive-metastore-service-chart/templates/configmap.yaml

```diff
@@ -2,28 +2,41 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: metastore-service-config
+  labels:
+    aissemble-configuration-store: enabled
 data:
   # Add all the default properties from the local values.yaml to the ConfigMap
   # Then check if there are any downstream properties and add them as well
   metastore-site.xml: |
+    {{- $basePropDict := dict }}
```

Comment:
S: Just use one dictionary. Write the base keys first, then write the props. That will automatically cause anything duplicate values in `properties` to take precedence over `baseProperties` without complex logic.

Reply:
Actually I'm not really sure why we need these dictionaries at all. Can you not just do `{{- if not hasKey .Values.configMap.metastoreServiceConfig.properties $baseProperty.name }}

Reply:
Sorry I'm bit new to helm syntax but from what I looked at the doc it has some limitation in creating variable and function but open to idea.

is hasProperty custom method? wasn't sure it was doable

Reply:
I can def cut follow up ticket if this requires refactoring given that this ticket is past DPC1!

Reply:
added https://jira.boozallencsn.com/browse/AIOPS-3838 as a follow up and will be working this